### ISSUE TITLE
mysql support, JenkinsBuidEnvinment satellite table added to data model

### ISSUE THREAD

**Issue by sergueik** [jenkinsci/audit2db-plugin#11]
re-adding accidentally closed pull request
Added mysql jdbc driver to the pom, updated pom to pull from http://repo.jenkins-ci.org/public/
Fixed defect in plugin causing packaging problem
Added JENKINS_BUILD_ENVIRONMENT model ( initial implementation : new database table, fields, reports) 
Fixed the test errors

**Comment by martinda** [comment]
Has this pull request been abandoned? I want to add postgresql to this plugin, so having mysql as an example to follow would be great! Thanks.

**Comment by sergueik** [comment]
Hello Martinda
Hope you are doing well. 

1.The original project owner has been quite busy *when*  I was active.
2. I was not been working on this code patch for quite some time due to other business that took priority. 


The patch to support additional driver jars is very elementary. I think that I submitted something else which was not of interest, and my social coding skills are not perfect... and hence I did not manage to get it mainstream.
Do you have capacity to compile the custom version of plugin on your cluster and use it ? I can assist with the patch then.  I did it in house and know  many people do the same due to some friction in code reviews.

If you want otherwise, it may need to wait , I am preparing to a round of job interviews recently, entirely different technology domain.

Thanks,
Serguei Kouzmine

**Comment by martinda** [comment]
No, I have not used the mysql version as it is the postgresql that I need to use in the end. I am able to compile the master branch here https://github.com/sergueik/audit2db-plugin. I see the [commits of the pull request](https://github.com/jenkinsci/audit2db-plugin/pull/10/commits), but there is a lot of white space change taking place so I find it hard to see where the mysql support was really added.

I also see that Kohsuke added a base plugin for database access https://github.com/jenkinsci/database-postgresql-plugin

At this point I try to decide which way to go.

**Comment by sergueik** [comment]
Hello Martinda

Thanks for looking.
I beleive the postgressql should be very similar to mysql in effort of adding. I can look into it ... 

Whitespace should not be a concern.
The feature I was adding is an additional hibrenate / persisstence class to collect envitonment of the job, but turns out while table is created the data is of moderate interest - I  am a heay user of envinhect plugin, this is why I  was doing this. I can segregate the changes. 

Let me know if you need assistance
Serguei Kouzmine

**Comment by sergueik** [comment]
Hello again Martinda
Let me iterate, the adding postgresql / mysql support is IIRC as easy as adding another pom dependency, with possibly no other code necessary, this is why i started with it. 

If you are confused by sorting through commits let me know if I have a window of time to look i will (but there are some other items on the list.
Thanks for your interestSerguei Kouzine
